percy/snapshot.py

```diff
@@ -55,6 +66,45 @@ def fetch_percy_dom():
     response.raise_for_status()
     return response.text
 
+@lru_cache(maxsize=None)
+def get_widths_for_multi_dom(widths):
+    # Deep copy mobile widths otherwise it will get overridden
+    allWidths = fetched_widths.get('mobile', [])[:]
+    if widths and len(widths) != 0:
+        allWidths.extend(widths)
+    else:
+        allWidths.extend(fetched_widths.get('config', []))
+    return list(set(allWidths))
+
+def change_window_dimension(driver, width, height):
+    if CDP_SUPPORT_SELENIUM and driver.capabilities['browserName'] == 'chrome':
+        driver.execute_cdp_cmd('Emulation.setDeviceMetricsOverride', { 'height': height,
+                               'width': width, 'deviceScaleFactor': 1, 'mobile': False })
+    else:
+        driver.set_window_size(width, height)
+
+def capture_responsive_dom(driver, cookies, **kwargs):
+    widths = kwargs.get('widths') or []
+    if 'width' in kwargs:
+        widths = [kwargs.get('width')]
+
+    # cache doesn't work when parameter is a list so passing tuple
+    widths = get_widths_for_multi_dom(tuple(widths))
+    dom_snapshots = []
+    window_size = driver.get_window_size()
+    current_width, current_height = window_size['width'], window_size['height']
+
+    for width in widths:
+        change_window_dimension(driver, width, current_height)
+        sleep(1)
```

Comment:
eh doesn't it mean we are slowing it by 1 sec. Use `wait` of selenium to ensure that command is over. Also command is sync in nature by default.

Reply:
page is already loaded so can't rely on page load / element visible event. i am adding wait so any change in DOM due to JS with take place. are you talking about selenium implicit wait?